pynicotine/gtkgui/widgets/theme.py

```diff
@@ -314,7 +314,6 @@ def set_global_style():
 
 CUSTOM_ICON_THEME_NAME = ".nicotine-icon-theme"
 FILE_TYPE_ICON_LABELS = {
-    "application-x-executable-symbolic": _("Executable"),
```

Comment:
What has this category been removed?

Reply:
There are too few file extensions, and I don't want to give special treatment to executables. You can just use "exe" as the filter if you want these files.

Reply:
A valid use for having both distinctive icon and an obvious method of filtering out executables does warrant special treatment, most users may be aware of the risk of such files but not all of them.

Reply:
Then the question becomes, what counts as an executable? Some of them don't have a file extension, and we have a bunch of package formats included in the archive-related extension list at the moment. If .exe is the main concern, it would be shorter to type than "executable" anyway.

I'm somewhat concerned that people would still (mis)use the filter to actually find the files, even if the intention is to exclude.

A distinctive icon is fine, but which extensions should have this treatment?

Reply:
Should we perhaps rename it to "Software" and include all package formats there?

Reply:
See "Potentially Dangerous File Extensions": https://www.lifewire.com/list-of-executable-file-extensions-2626061

Things like `bat` `scr`, perhaps even scripts like `py` etc.

Reply:
![](https://user-images.githubusercontent.com/8754153/218254600-6ec08590-86eb-4772-a4db-5173817a1e2a.png)

Reply:
> Maybe Software is too generic

"Program", "Package", "Installer"

Reply:
On second thought, I think "Executable" drives the point home faster (exe, executable, can be run, possibly dangerous).

Reply:
> Isn't there a library we can use to match extensions with mime types known by the OS?

Perhaps https://docs.python.org/3/library/mimetypes.html

Reply:
Audio, image and video work fine with the mimetypes module, since these mimetypes start with `audio/`, `image/` and `video/`.

Everything else falls apart, since it either starts with `application/`, or includes a lot of junk we don't want to show (`text/`, which includes e.g. json, xml, config files, source files, which is most certainly junk on the Soulseek network).

Reply:
Seems like the list is also manually updated on other OSes than Windows, so some extensions will be missing in older Python versions: https://github.com/python/cpython/commits/3.11/Lib/mimetypes.py

I think I'll stick with our curated lists for now.